execute/exectypes/outcome.go

```diff
@@ -7,8 +7,40 @@ import (
 	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
 )
 
+type PluginState int
```

Comment:
I wonder if it would be useful to have a `MarshalJSON` on this so that the outcomes that are JSON encoded are still easily understandable (due to a human-readable string rather than a number). I think similar logic applies to the commit states

Reply:
Maybe. I would personally prefer if these had a more efficient encoding rather than attempting to make them human readable.

In this case the states would be 1->2->3, so still pretty human readable.

Reply:
I may be unlike others when it comes to this but even just 1, 2 and 3 is already a cognitive jump because I have to continuously remind myself what each number means.

Having debugged the v2 plugins quite a bit I'd rather we optimize for debuggability to start and then come back and make encoding tighter if needed.

Reply:
OK. Changed this type to a string.

Reply:
Also worth noting that we have states for commit as well, with their own integers, and states in many other systems which have their own integers, etc. Having all of this in mind I feel like the actual string encoding is just better and it doesn't really have that much overhead to outcome encoding size

Reply:
The phases could also be named "ReadDest", "ReadSource" and "ReadDestAgain"

Reply:
Less descriptive I feel, I kinda like the current names better, but not sure what other folks think. @asoliman92?